ml-agents/mlagents/trainers/trajectory.py

```diff
@@ -107,12 +107,15 @@ def to_agentbuffer(self) -> AgentBuffer:
             agent_buffer_trajectory["done"].append(exp.done)
             # Add the outputs of the last eval
             if exp.action_pre is not None:
-                actions_pre = exp.action_pre
-                agent_buffer_trajectory["actions_pre"].append(actions_pre)
+                agent_buffer_trajectory["actions_pre"].append(exp.action_pre)
 
-            # value is a dictionary from name of reward to value estimate of the value head
-            agent_buffer_trajectory["actions"].append(exp.action)
-            agent_buffer_trajectory["action_probs"].append(exp.action_probs)
+            # Adds the log prob and action of continuous/discrete separately
+            action_shape = None
+            for act_type, act_array in exp.action.items():
+                agent_buffer_trajectory[act_type].append(act_array)
+                action_shape = act_array.shape  # TODO Better way to make mask
```

Comment:
It seems like this relies on the fact that discrete actions are last in the dict. I'd make it explicit so that if the order of the dict changes it won't break.